File: rulepacks/2026.08.3/testdata/datos-ext.go
// Fixture de semgrep --test — no es código del producto.
// Casos de prueba del pilar datos (stem datos-ext), lenguaje Go.
package fixtures

import (
	"fmt"
	"log"
)

// Repositorio simula un ORM mínimo para los casos de transacción.
type Repositorio struct{}

func (r *Repositorio) Create(v interface{}) {}

func (r *Repositorio) Update(v interface{}) {}

func (r *Repositorio) Begin() *Repositorio { return r }

func (r *Repositorio) Commit() {}

// --- go-dinero-float ------------------------------------------------------------
// Factura modela un cobro del corpus de pruebas.
type Factura struct {
	// ruleid: go-dinero-float
	ImporteTotal float64
	// ok: go-dinero-float
	Latitud float64
}

// --- log-dato-sensible ----------------------------------------------------------
func registrarSesion(token string) {
	// ruleid: log-dato-sensible
	log.Printf("token emitido: %s", token)
	// ok: log-dato-sensible
	fmt.Println("token restringido: consulte la ayuda del comando")
}

// --- escrituras-sin-transaccion-go ------------------------------------------------
func guardarPedido(conn *Repositorio, pedido, reserva interface{}) {
	// ruleid: escrituras-sin-transaccion-go
	conn.Create(pedido)
	conn.Update(reserva)
}

func guardarPedidoAtomico(db *Repositorio, pedido, reserva interface{}) {
	tx := db.Begin()
	// ok: escrituras-sin-transaccion-go
	tx.Create(pedido)
	tx.Update(reserva)
	tx.Commit()
}

File: rulepacks/2026.08.3/testdata/datos-ext.cs
// Fixture de semgrep --test — no es código del producto.
// Casos de prueba del pilar datos (stem datos-ext), lenguaje C#.
using System;
using System.Collections.Generic;
using System.Linq;

namespace Fixtures
{
    public class DatosExt
    {
        // --- csharp-dinero-float ---------------------------------------------
        public void CalcularCobro()
        {
            // ruleid: csharp-dinero-float
            double montoTotal = 100.0;
            // ok: csharp-dinero-float
            double distanciaKm = 12.5;
        }

        // --- csharp-datetime-naive -------------------------------------------
        public void MarcarTiempos()
        {
            // ruleid: csharp-datetime-naive
            var creado = DateTime.Now;
            // ok: csharp-datetime-naive
            var registrado = DateTimeOffset.UtcNow;
        }

        // --- csharp-orm-en-bucle ---------------------------------------------
        public void CargarClientes(MiContexto ctx, List<int> ids)
        {
            foreach (int id in ids)
            {
                // ruleid: csharp-orm-en-bucle
                var cliente = ctx.Clientes.FirstOrDefault(c => c.Id == id);
            }
            // ok: csharp-orm-en-bucle
            var primero = ctx.Clientes.FirstOrDefault();
        }
    }
}
